internal/kubernetes/operator/install_resources.go

```diff
@@ -113,32 +127,32 @@ func (ir *InstallResources) InstallConfiguration(ctx context.Context, version, n
 	for _, config := range configData {
 		switch config["kind"] {
 		case "ServiceAccount":
-			err = ir.addServiceAccount(ctx, config, namespace)
+			err = ir.addServiceAccount(ctx, config, installConfig.Namespace)
 			if err != nil {
 				return err
 			}
 		case "Role":
-			err = ir.addRoles(ctx, config, namespace, watch)
+			err = ir.addRoles(ctx, config, installConfig.Namespace, installConfig.Watch)
 			if err != nil {
 				return err
 			}
 		case "ClusterRole":
-			err = ir.addClusterRole(ctx, config, namespace)
+			err = ir.addClusterRole(ctx, config, installConfig.Namespace)
 			if err != nil {
 				return err
 			}
 		case "RoleBinding":
-			err = ir.addRoleBindings(ctx, config, namespace, watch)
+			err = ir.addRoleBindings(ctx, config, installConfig.Namespace, installConfig.Watch)
 			if err != nil {
 				return err
 			}
 		case "ClusterRoleBinding":
-			err = ir.addClusterRoleBinding(ctx, config, namespace)
+			err = ir.addClusterRoleBinding(ctx, config, installConfig.Namespace)
 			if err != nil {
 				return err
 			}
 		case "Deployment":
-			err = ir.addDeployment(ctx, config, namespace, watch, atlasGov)
+			err = ir.addDeployment(ctx, config, installConfig.Namespace, installConfig.Watch, installConfig.ResourceDeletionProtectionEnabled, installConfig.SubResourceDeletionProtectionEnabled, installConfig.AtlasGov)
```

Comment:
This function uses the entire struct except the version attribute, what do you think about passing the entire struct instead of each attribute?

Reply:
Fixed!